crypto/evp/p_lib.c

```diff
@@ -748,6 +748,14 @@ const void *EVP_PKEY_get0(const EVP_PKEY *pkey)
     return evp_pkey_get_legacy((EVP_PKEY *)pkey);
 }
 
+void *EVP_PKEY_get0_legacy(EVP_PKEY *pkey)
+{
+    if (pkey == NULL)
+        return NULL;
+
+    return evp_pkey_get_legacy((EVP_PKEY *)pkey);
```

Comment:
Probably this should just return pkey->pkey.ptr so that a NULL is returned in the event that this a non-legacy key.